app/sagas/videoConf.ts

```diff
@@ -0,0 +1,243 @@
+import { getSubscriptionByRoomId } from '../lib/database/services/Subscription';
+import { appSelector } from '../lib/hooks';
+import { callJitsi } from '../lib/methods';
+import { compareServerVersion, showErrorAlert } from '../lib/methods/helpers';
+import log from '../lib/methods/helpers/log';
+import { hideCustomNotification, showCustomNotification } from '../lib/methods/helpers/notifications';
+import { showToast } from '../lib/methods/helpers/showToast';
+import { videoConfJoin } from '../lib/methods/videoConf';
+import { Services } from '../lib/services';
+import { notifyUser } from '../lib/services/restApi';
+import { ICallInfo } from '../reducers/videoConf';
+
+interface IGenericAction extends Action {
+	type: string;
+}
+
+type THandleGeneric = IGenericAction & {
+	data: any;
+};
+
+type TInitCallGeneric = IGenericAction & {
+	payload: TCallProps;
+};
+
+type TCancelCallGeneric = IGenericAction & {
+	payload?: { callId?: string };
+};
+
+type TAcceptCallGeneric = IGenericAction & {
+	payload: { callId: string };
+};
+
+// The interval between attempts to call the remote user
+const CALL_INTERVAL = 3000;
+// How many attempts to call we're gonna make
+const CALL_ATTEMPT_LIMIT = 10;
+
+function* onDirectCall(payload: ICallInfo) {
+	const calls = yield* appSelector(state => state.videoConf.calls);
+	const currentCall = calls.find(c => c.callId === payload.callId);
+	const hasAnotherCall = calls.find(c => c.action === 'call');
+	if (hasAnotherCall && hasAnotherCall.callId !== payload.callId) {
+		return;
+	}
+	if (!currentCall) {
+		yield put(setVideoConfCall(payload));
+		showCustomNotification(IncomingCallComponent, payload, 30000);
+	}
+}
+
+function* onDirectCallCanceled(payload: ICallInfo) {
+	const calls = yield* appSelector(state => state.videoConf.calls);
+	const currentCall = calls.find(c => c.callId === payload.callId);
+	if (currentCall) {
+		yield put(removeVideoConfCall(currentCall));
+		hideCustomNotification();
+	}
+}
+
+function* onDirectCallAccepted({ callId, rid, uid, action }: ICallInfo) {
+	const calls = yield* appSelector(state => state.videoConf.calls);
+	const userId = yield* appSelector(state => state.login.user.id);
+	const currentCall = calls.find(c => c.callId === callId);
+	if (currentCall && currentCall.action === 'calling') {
+		yield call(notifyUser, `${uid}/video-conference`, { action: 'confirmed', params: { uid: userId, rid, callId } });
+		yield put(setVideoConfCall({ callId, rid, uid, action }));
+	}
+}
+
+function* onDirectCallRejected() {
+	yield call(cancelCall, {});
+	showToast(i18n.t('Call_rejected'));
+	yield call(hideActionSheetRef);
+}
+
+function* onDirectCallConfirmed(payload: ICallInfo) {
+	const calls = yield* appSelector(state => state.videoConf.calls);
+	const currentCall = calls.find(c => c.callId === payload.callId);
+	if (currentCall) {
+		yield put(removeVideoConfCall(currentCall));
+		yield call(hideActionSheetRef);
+		videoConfJoin(payload.callId, false, true);
```

Comment:
I think you should `yield` and `try/catch` this.
Review the other usages of `videoConfJoin`.
Seeing the same issues on some components.